File: auction_service/src/main/java/com/example/auction/model/dao/AuctionEntity.java
package com.example.auction.model.dao;

import lombok.Data;
import lombok.Getter;

import javax.persistence.*;
import java.time.LocalDate;
import java.util.List;

@Entity
@Table(name = "auctions")
@Data
public class AuctionEntity {

    @Id
    @GeneratedValue(strategy= GenerationType.IDENTITY)
    private Long id;

    private String title;

    private String description;

    @OneToMany(cascade = {CascadeType.ALL})
    @JoinColumn(name="auction_id")
    private List<ImageEntity> pictures;

    @ManyToOne
    @JoinColumn
    private CategoryEntity category;

    private double minPrice;

    private double buyNowPrice;

    private boolean promoted;

    @OneToOne(cascade = {CascadeType.ALL})
    private LocalizationEntity localization;

    private LocalDate dateOfStart;

    private LocalDate dateOfFinish;

    private int amountOfVisits;

    @ManyToOne
    @JoinColumn
    private UserEntity userEntity;

    private boolean isActive;


}

File: auction_service frontend/auction/src/app/AuctionEntity.ts

import { LocalizationEntity } from "./LocalizationEntity";



export class AuctionEntity {

  private title: String;
  private category: String;
  private description: String;
  private minPrice: number;
  private buyNowPrice: number;
  private address: LocalizationEntity;
  private promoted: boolean;
  private images: String[];
  private emailAddress: String;


  constructor(title: String, category: String, description: String, minPrice: number, buyNowPrice: number, address: LocalizationEntity, promoted: boolean, images: String[], emailAddress: String) {
    this.title = title;
    this.category = category;
    this.description = description;
    this.minPrice = minPrice;
    this.buyNowPrice = buyNowPrice;
    this.address = address;
    this.promoted = promoted;
    this.images = images;
    this.emailAddress = emailAddress

  }
}